In [ ]:
from google.colab import drive

drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
!pip install fastparquet

import pandas as pd

pd.set_option('display.max_columns', None) # show alll columns

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.5/1.5 MB 6.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 12.9 MB/s eta 0:00:00


-------------------

# review

-------------------

In [ ]:
df = pd.read_parquet('/content/drive/MyDrive/DataYelpGoogle/reviews.parquet', engine='fastparquet')

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6990280 entries, 0 to 6990279
Data columns (total 9 columns):
 #   Column       Dtype  
---  ------       -----  
 0   review_id    object 
 1   user_id      object 
 2   business_id  object 
 3   stars        float64
 4   useful       int64  
 5   funny        int64  
 6   cool         int64  
 7   text         object 
 8   date         object 
dtypes: float64(1), int64(3), object(5)
memory usage: 480.0+ MB


In [ ]:
# cambiar tipo de dato
df.stars = df.stars.astype('int32')
df.useful = df.useful.astype('int32')
df.funny = df.funny.astype('int32')
df.cool = df.cool.astype('int32')
df.date = pd.to_datetime(df['date'], format='%Y-%m-%d')

In [ ]:
# show data
df.sample(5)

review_id                 user_id  \
6434843  LLIFYd-0CZuU2m_ajd_CYw  UTpuSgxBuCUQFLRfIQQMiQ   
6750226  aOTZ1zFUdP2HFw98i-ufpw  q39mVpfBW8GQ57sjyJRwNA   
1556650  DBtyt974OjrjX3OIhtmj3Q  9rQ8G8Bx7StuuAtkLXesKQ   
4472762  r-BDXvQLtOoqXQwGHuQIFg  f-9AtOIvpbDrGOYJxx0pDw   
381125   hJlXCnacEAy6Hkj_Zo0cKQ  ZK7DQBXtTDROUvgA9P_nXQ   

                    business_id  stars  useful  funny  cool  \
6434843  TOGGCJcfK2KY8_Pnjrk5FA      4       1      1     0   
6750226  HJQMHwCpLe0m1BeCTEbn3w      2       2      0     0   
1556650  wB1Tin0OW1JRpaKM-E3ZYA      2       0      0     0   
4472762  LnbiHKW-cOFpigCdK6Zz9A      5       0      0     1   
381125   2t16TUHzPImXJjfl2fX_dA      1       1      0     0   

                                                      text                date  
6434843  Four stars because cheap booze.\n\n$3 gets you... 2014-01-27 20:23:31  
6750226  I travelled some distance to try this place af... 2018-06-18 02:11:52  
1556650  Had a Blue Moon draft, it was not very cold...... 2017-07-03 22:37:10  
4472762  While I wish I was smart enough to change the ... 2013-12-18 23:55:25  
381125   My new employer recommended Concentra for a re... 2019-07-05 17:05:40

In [ ]:
# cantidad nulos
df.isna().sum()

review_id      0
user_id        0
business_id    0
stars          0
useful         0
funny          0
cool           0
text           0
date           0
dtype: int64

In [ ]:
# exploracion variable numericas
df.describe()

stars        useful         funny          cool
count  6.990280e+06  6.990280e+06  6.990280e+06  6.990280e+06
mean   3.748584e+00  1.184609e+00  3.265596e-01  4.986175e-01
std    1.478705e+00  3.253767e+00  1.688729e+00  2.172460e+00
min    1.000000e+00 -1.000000e+00 -1.000000e+00 -1.000000e+00
25%    3.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00
50%    4.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00
75%    5.000000e+00  1.000000e+00  0.000000e+00  0.000000e+00
max    5.000000e+00  1.182000e+03  7.920000e+02  4.040000e+02

In [ ]:
# fecha minima
df['date'].min()

Timestamp('2005-02-16 03:23:22')

In [ ]:
# fecha maxima
df['date'].max()

Timestamp('2022-01-19 19:48:45')

In [ ]:
# crear columna year
df['year'] = df['date'].dt.year

In [ ]:
# cambiar tipo de dato
df.year = df.year.astype('int32')

In [ ]:
# filtrar año mayor igual 2017
df = df[df['year']>=2017]
df.reset_index(drop=True, inplace=True) # reset index

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3838105 entries, 0 to 3838104
Data columns (total 10 columns):
 #   Column       Dtype         
---  ------       -----         
 0   review_id    object        
 1   user_id      object        
 2   business_id  object        
 3   stars        int32         
 4   useful       int32         
 5   funny        int32         
 6   cool         int32         
 7   text         object        
 8   date         datetime64[ns]
 9   year         int32         
dtypes: datetime64[ns](1), int32(5), object(4)
memory usage: 219.6+ MB


In [ ]:
# show data
df.sample(5)

review_id                 user_id  \
3128831  Bjjzjpzhi71FIkbE_nib2Q  iOfnM6E8HTYs9YqGLT-NWg   
753012   AdP909irPqO1nMMcQaXFIA  5sS0OgV1pKHA0qdf2-lmqg   
1723717  IvGK7AW85DJT59XUhWZNJA  ZzoQFtNu4qfn2epgko7DaQ   
1826959  krxMAgnfu0dvp9s6f2d7lg  1pT3x92oLtKcJY-eyW7fhw   
1628152  xc36w4zk8GddzQTePyRbfw  RtKhjGi_CMQYCnZRy_R-QQ   

                    business_id  stars  useful  funny  cool  \
3128831  zW37VoPxHHTTWTIGNWjZgg      5       0      0     1   
753012   v7DDirUL-ZA-CmRgl7m-pg      5       3      3     2   
1723717  CgiwCrXs3Ggu9eKFPIgjtA      4       0      0     0   
1826959  khJNtyRfPwQ0ybr3tPeZ3g      5       4      0     3   
1628152  Xh3PAJG65l92cW8Z3gQ2VA      1       0      0     1   

                                                      text  \
3128831  Stopped here today on our way from Illinois to...   
753012   Phenomenal soul food! I had the oxtail meal an...   
1723717  We went on a Monday night, and it was packed! ...   
1826959  I think they put magic dust on the sweet potat...   
1628152  Pizza was cold and the crust was awful, and it...   

                       date  year  
3128831 2017-04-17 02:34:18  2017  
753012  2021-03-24 01:07:20  2021  
1723717 2019-07-09 20:04:49  2019  
1826959 2019-02-28 23:38:48  2019  
1628152 2017-05-11 05:41:49  2017

In [ ]:
# crear parquet filtro año mayor igual 2017
df.to_parquet('reviewB.parquet', index=False)

# review - año 2017 - 2022

In [ ]:
df = pd.read_parquet('/content/drive/MyDrive/DataYelpGoogle/reviewB.parquet', engine='fastparquet')

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3838105 entries, 0 to 3838104
Data columns (total 10 columns):
 #   Column       Dtype         
---  ------       -----         
 0   review_id    object        
 1   user_id      object        
 2   business_id  object        
 3   stars        int32         
 4   useful       int32         
 5   funny        int32         
 6   cool         int32         
 7   text         object        
 8   date         datetime64[ns]
 9   year         int32         
dtypes: datetime64[ns](1), int32(5), object(4)
memory usage: 219.6+ MB


In [ ]:
# show data
df.sample(10)

review_id                 user_id  \
2875955  2MDM_6a-9PWOG_RpNWwwlg  ec-Z7VOATcpi-F7H40C2oQ   
2132719  iTs64778NxzGDzxDOMn5Dg  vQ1OOx4NELx9LUB7JWG2HA   
866111   Vk32EsfHdZRIs8pLiDODog  8xNsJ8Amji8xhaQ-3ABZ-w   
1930165  3L4PK-2b_ojX9E4OJuPykA  gYtmlceUcCFtJf75YVDJkA   
501351   HfDmi8ruZ2uyn-p4JFUWog  BhVYeXN3H3rvmpIb9dvBMg   
1584472  053hxY-Pl4bZjiguN8-6hg  c6Wtbu50bGWqaczYXe_Cnw   
2470991  g84cUXElUwp5npSqSzjGsg  zfDG3229XSu5xZYc-AZNvw   
3683308  NldbxsV9OtYzjiY1LHK8SA  U-uR_N-WhmHZ4edKwLh01w   
3082231  kaN9SSQMElx8dR9708dbkw  071rPDxb9RqEMlONBckQVw   
1785573  9hefHCHwDBEEHtuV7_Z4Pg  kz55Wcoqa96p8c4T51LVHQ   

                    business_id  stars  useful  funny  cool  \
2875955  vznHFDZ8-oZbJRU6Bslsdw      5       1      0     1   
2132719  9fuSR8RN6E6T20DNVSqZ8g      3       3      0     0   
866111   rnfgkc6B_5O3NOKXYqJaZA      5       0      0     0   
1930165  UR9dzGGVfcJhHt_mG3bTRA      5       0      0     0   
501351   h81N-kVGvV9f_PWJQJN9YA      5       0      0     0   
1584472  GKcaAbGsobMVhjSFktgVsA      5       0      0     0   
2470991  JjYmXia9-Dy0tGqI088Hfg      5       0      0     0   
3683308  Gr0rLHeoUQC1S6412FZhoQ      1       0      0     0   
3082231  fxV5Iv7goPEUz-AReRLxkw      4       0      0     0   
1785573  mdlWkVgVv7Tgn3OlZGUxIQ      5       1      0     0   

                                                      text  \
2875955  Let me start off by saying yes....they are a n...   
2132719  I believe the biggest issue with this nail sal...   
866111   Very good smoothies. Great staff very helpful ...   
1930165  Excellent food; excellent service as well! We ...   
501351   This is the second time I come here and I have...   
1584472  My wife and I went out for our New Years Eve c...   
2470991  I was in the Berwyn area for the first time in...   
3683308  I wish I could give them zero stars. Showed up...   
3082231  The food was great, very large portions. Defin...   
1785573  Food is delicious! Beautifully prepared, and s...   

                       date  year  
2875955 2019-02-18 02:33:19  2019  
2132719 2018-05-17 00:29:34  2018  
866111  2018-11-09 01:28:22  2018  
1930165 2017-10-15 01:40:55  2017  
501351  2018-04-16 23:25:58  2018  
1584472 2018-01-01 19:30:31  2018  
2470991 2019-05-09 21:46:33  2019  
3683308 2020-06-08 16:43:33  2020  
3082231 2018-03-25 15:12:50  2018  
1785573 2020-10-22 19:53:06  2020

In [ ]:
# cantidad nulos
df.isna().sum()

review_id      0
user_id        0
business_id    0
stars          0
useful         0
funny          0
cool           0
text           0
date           0
year           0
dtype: int64

In [ ]:
# exploracion variable numericas
df.describe()

stars        useful         funny          cool          year
count  3.838105e+06  3.838105e+06  3.838105e+06  3.838105e+06  3.838105e+06
mean   3.764416e+00  9.710623e-01  2.398335e-01  4.703610e-01  2.018828e+03
std    1.547572e+00  2.938497e+00  1.409925e+00  2.272291e+00  1.387146e+00
min    1.000000e+00 -1.000000e+00 -1.000000e+00 -1.000000e+00  2.017000e+03
25%    3.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  2.018000e+03
50%    5.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  2.019000e+03
75%    5.000000e+00  1.000000e+00  0.000000e+00  0.000000e+00  2.020000e+03
max    5.000000e+00  9.330000e+02  7.920000e+02  3.040000e+02  2.022000e+03

In [ ]:
# duplicados?
df[df.duplicated()]

Empty DataFrame
Columns: [review_id, user_id, business_id, stars, useful, funny, cool, text, date, year]
Index: []

In [ ]:
# valores unicos review
df['review_id'].nunique()

3838105

In [ ]:
# cantidad de usuarios q realizaron al menos una review
df['user_id'].nunique()

1357278

In [ ]:
# cantidad de business valorados
df['business_id'].nunique()

135542

In [ ]:
# agrupar numero de reviews por año
df.groupby(['year'])['review_id'].count().sort_values()

year
2022     31665
2020    554557
2021    618189
2017    820048
2018    906362
2019    907284
Name: review_id, dtype: int64

In [ ]:
# fecha maxima
df['date'].max()

Timestamp('2022-01-19 19:48:45')

considerar que los datos del año 2022 son solo hasta el 19 de enero, por tal motivo las pocas reseñas en dicho año

In [ ]:
# agrupar numero de reviews por rating
round(df.groupby(['stars'])['review_id'].count().apply(lambda x: (x/3838105)*100).sort_values(ascending=False), 2)

stars
5    51.17
1    17.50
4    16.18
3     8.07
2     7.08
Name: review_id, dtype: float64

calificaciones con 5 estrellas 51.17% del total de los reviews encabeza la lista seguido de las calificaciones con 1 estrella con 17.50% del total de los review. Dato interesante

In [ ]:
df['useful'].min()

-1

In [ ]:
df[df['useful']<0]

review_id                 user_id  \
1005783  plhadSWfXZFKRAhEztb6Dw  0Q4ouqkcwrVE_Im6a6Tomw   
1101632  iYS4pfUCUPT_TEmW6BOnrA  1p8Pu1pvAsuNtuZDtn2OkQ   
1191162  JcWLycg0Hdsz6LBNAqS37A  04j3r2Psfw_LxX1UXPhnyg   

                    business_id  stars  useful  funny  cool  \
1005783  SJQePJL1bkeEk-6gp-6eeA      1      -1      0     0   
1101632  SJQePJL1bkeEk-6gp-6eeA      3      -1      0     0   
1191162  E3Qk7xTznkT0SABL2ltpHA      5      -1     -1    -1   

                                                      text  \
1005783  This complaint has nothing to do with the qual...   
1101632  Having been to bouncy places across the countr...   
1191162  The food was so much better than I expected. T...   

                       date  year  
1005783 2018-12-23 01:35:31  2018  
1101632 2017-01-18 18:11:15  2017  
1191162 2017-10-05 02:47:33  2017

In [ ]:
df[df['funny']<0]

review_id                 user_id  \
1191162  JcWLycg0Hdsz6LBNAqS37A  04j3r2Psfw_LxX1UXPhnyg   
2351781  7tlF2NEDiFqMRjKBgTPJ3g  VKK7pPl6DfjL9H5b3n2yCA   

                    business_id  stars  useful  funny  cool  \
1191162  E3Qk7xTznkT0SABL2ltpHA      5      -1     -1    -1   
2351781  qjstXewt23oCZnf86CsHXg      1       2     -1     2   

                                                      text  \
1191162  The food was so much better than I expected. T...   
2351781  The most horrible removal company I have ever ...   

                       date  year  
1191162 2017-10-05 02:47:33  2017  
2351781 2017-02-25 23:40:34  2017

In [ ]:
df[df['cool']<0]

review_id                 user_id  \
1191162  JcWLycg0Hdsz6LBNAqS37A  04j3r2Psfw_LxX1UXPhnyg   

                    business_id  stars  useful  funny  cool  \
1191162  E3Qk7xTznkT0SABL2ltpHA      5      -1     -1    -1   

                                                      text  \
1191162  The food was so much better than I expected. T...   

                       date  year  
1191162 2017-10-05 02:47:33  2017

solo 4 registros del total de los datos con valor atipico [useful, funny, cool] = -1

In [ ]:
# cantidad reviews agrupadas por calificacion y año
df.groupby(['year', 'stars'])['review_id'].count()

year  stars
2017  1        127385
      2         61736
      3         75147
      4        156578
      5        399202
2018  1        148425
      2         66025
      3         77119
      4        158766
      5        456027
2019  1        157311
      2         64658
      3         74458
      4        146931
      5        463926
2020  1        105401
      2         35000
      3         37011
      4         74469
      5        302676
2021  1        126450
      2         42494
      3         43687
      4         80063
      5        325495
2022  1          6607
      2          1959
      3          2183
      4          4351
      5         16565
Name: review_id, dtype: int64

In [ ]:
# crear columna hour
df['hour'] = df['date'].dt.hour

In [ ]:
# cantidad reviews agrupadas por hora del dia
df.groupby(['hour'])['review_id'].count().sort_values(ascending=False)

hour
0     265772
23    258700
19    258640
1     258474
18    258002
20    252276
22    249098
21    246351
17    241215
2     228682
16    216758
15    188176
3     180031
14    157150
4     127285
13    119157
5      80431
12     76696
6      48299
11     41760
7      28534
10     21946
8      18433
9      16239
Name: review_id, dtype: int64

se identifica mayor cantidad de review a media noche. Dato interesante

In [ ]:
# cantidad de reviews para cada business
df['business_id'].value_counts().head()

ac1AeYqs8Z4_e2X5M3if2A    5161
GXFMD0Z4jEVZBCsbPf4CTQ    3811
_ab50qdWOk0DdB6XOrBitw    3669
oBNrLz4EDhiscSlbOl8uAw    3494
1b5mnK8bMnnju_cvU65GqQ    3147
Name: business_id, dtype: int64

In [ ]:
# cantidad de reviews por usuario
df['user_id'].value_counts().head()

wXdbkFZsfDR7utJvbWElyA    1206
fr1Hz2acAb3OaL3l6DyKNg     888
-G7Zkl1wIWBBmD0KRy_sCw     775
vmUqcqMjlWoBM6qfmUXgyQ     772
Sp2GV7D-_JLZMPQmDanzPQ     723
Name: user_id, dtype: int64

In [ ]:
# review con mayor cantidad de useful votes
df.sort_values(by='useful', ascending=False).head(5)

review_id                 user_id  \
3547183  TFlWosaJn3lVJwgm3nrVYA  9YOVI-bxiiOx0bTZ-J_Irw   
3608182  wEHqtpyPdHGFXDEqmmtEkQ  GIDIA7h46PDp2LirqCjtpw   
225090   71HvnEk9a3SVnPO5spQBSg  uV4edOcsZ1MuCmOUSM41Xg   
317250   D2UGrB5A0RRFTF9lLS94uQ  6hftNZ-rXnr1N9rlzIlQqg   
3803169  HnnDqF0ksknHEy7B2Im9Fg  giJqvuH8EC62Hi5WboZ08w   

                    business_id  stars  useful  funny  cool  \
3547183  3RcQ7yTvJn9LLuXthTkWiQ      1     933      0     0   
3608182  3RcQ7yTvJn9LLuXthTkWiQ      2     617      0     0   
225090   CkxfUzD10QVOsdWQDucJ6w      1     539     26    20   
317250   CkxfUzD10QVOsdWQDucJ6w      1     346      8    10   
3803169  1b5mnK8bMnnju_cvU65GqQ      5     325    180   304   

                                                      text  \
3547183  If you are unfortunate enough to have to deal ...   
3608182  I'm a current customer.  Ditech would not post...   
225090   This woman is an absolute lunatic. Stay away. ...   
317250   Not only was the quality and selection poor, b...   
3803169  Who better to fall in sweet Biscuit Love with ...   

                       date  year  
3547183 2017-01-13 18:26:56  2017  
3608182 2017-02-28 23:18:31  2017  
225090  2021-04-12 03:48:38  2021  
317250  2021-01-02 17:41:10  2021  
3803169 2019-02-22 21:27:43  2019

los usuarios votan useful una review con mayor frecuencia a valoraciones por debajo del promedio. Los usuarios buscan con mayor frecuencia reviews en las cuales se hable de un problema que se tuvo con el negocio para evitar cometer errores

In [ ]:
# review con mayor cantidad de funny votes
df.sort_values(by='funny', ascending=False).head(5)

review_id                 user_id  \
3750456  pL_jmXOSPOMq71ZwrlRynQ  sMfFISqyKkxlLsRzYaL9Lw   
347425   q6_sDa8rUZsaxUbTApWfjA  Cf9DiXFtnP7N_5IjyyKMZQ   
2261388  7d3KH3BOdQjFarD_a0NQUA  JjXuiru1_ONzDkYVrHN0aw   
3803169  HnnDqF0ksknHEy7B2Im9Fg  giJqvuH8EC62Hi5WboZ08w   
1839347  4TAKLBW49b9DFFuGatGkLA  JjXuiru1_ONzDkYVrHN0aw   

                    business_id  stars  useful  funny  cool  \
3750456  iRIHK8-EwpeffwvoO4nzIA      1     112    792   107   
347425   Zx7n8mdt8OzLRXVzolXNhQ      3       0    254     0   
2261388  m1HVolBJiYajyq07J550jQ      5     262    201   261   
3803169  1b5mnK8bMnnju_cvU65GqQ      5     325    180   304   
1839347  VVH6k9-ycttH3TV_lk5WfQ      4     222    159   207   

                                                      text  \
3750456  Went there for a birthday dinner and had reser...   
347425   Menu reads better than performance of food. Ha...   
2261388  REVIEW # 1,000\n\nIn 2012 I began yelping regu...   
3803169  Who better to fall in sweet Biscuit Love with ...   
1839347  This is my third visit to New Orleans (2nd as ...   

                       date  year  
3750456 2017-02-19 13:26:19  2017  
347425  2018-05-12 21:02:13  2018  
2261388 2017-12-18 18:10:10  2017  
3803169 2019-02-22 21:27:43  2019  
1839347 2017-12-10 20:24:28  2017

In [ ]:
# review con mayor cantidad de cool votes
df.sort_values(by='cool', ascending=False).head(5)

review_id                 user_id  \
3803169  HnnDqF0ksknHEy7B2Im9Fg  giJqvuH8EC62Hi5WboZ08w   
2261388  7d3KH3BOdQjFarD_a0NQUA  JjXuiru1_ONzDkYVrHN0aw   
1839347  4TAKLBW49b9DFFuGatGkLA  JjXuiru1_ONzDkYVrHN0aw   
2592399  ehUVIpfylhrxraozNY_TDQ  JjXuiru1_ONzDkYVrHN0aw   
3017751  HA0ZLe6xoyRIkrGQ8areXw  K74S1UK7QjAhWnkXIN0nhg   

                    business_id  stars  useful  funny  cool  \
3803169  1b5mnK8bMnnju_cvU65GqQ      5     325    180   304   
2261388  m1HVolBJiYajyq07J550jQ      5     262    201   261   
1839347  VVH6k9-ycttH3TV_lk5WfQ      4     222    159   207   
2592399  SNxUXrSr-XkEkSQIdjmjhA      3     227    159   207   
3017751  J0joPXxmN-_9Lzafspqdbw      4     209    110   205   

                                                      text  \
3803169  Who better to fall in sweet Biscuit Love with ...   
2261388  REVIEW # 1,000\n\nIn 2012 I began yelping regu...   
1839347  This is my third visit to New Orleans (2nd as ...   
2592399  I've been coming to Hank's Catfish & BBQ for m...   
3017751  An American diner with a mixture of Hawaiian a...   

                       date  year  
3803169 2019-02-22 21:27:43  2019  
2261388 2017-12-18 18:10:10  2017  
1839347 2017-12-10 20:24:28  2017  
2592399 2018-12-10 18:16:28  2018  
3017751 2019-10-09 09:27:26  2019

-------------------

# user

-------------------

In [ ]:
df = pd.read_parquet('/content/drive/MyDrive/DataYelpGoogle/user.parquet', engine='fastparquet')

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2105597 entries, 0 to 2105596
Data columns (total 22 columns):
 #   Column              Dtype  
---  ------              -----  
 0   user_id             object 
 1   name                object 
 2   review_count        int64  
 3   yelping_since       object 
 4   useful              int64  
 5   funny               int64  
 6   cool                int64  
 7   elite               object 
 8   friends             object 
 9   fans                int64  
 10  average_stars       float64
 11  compliment_hot      int64  
 12  compliment_more     int64  
 13  compliment_profile  int64  
 14  compliment_cute     int64  
 15  compliment_list     int64  
 16  compliment_note     int64  
 17  compliment_plain    int64  
 18  compliment_cool     int64  
 19  compliment_funny    int64  
 20  compliment_writer   int64  
 21  compliment_photos   int64  
dtypes: float64(1), int64(16), object(5)
memory usage: 353.4+ MB


In [ ]:
# cambiar tipo de datos
df.review_count = df.review_count.astype('int32')
df.yelping_since = pd.to_datetime(df['yelping_since'], format='%Y-%m-%d')
df.useful = df.useful.astype('int32')
df.funny = df.funny.astype('int32')
df.cool = df.cool.astype('int32')
df.fans = df.fans.astype('int32')
df.average_stars = df.average_stars.astype('float32')

In [ ]:
# show data
df.sample(5)

user_id    name  review_count       yelping_since  \
2077614  AM9isBV9lO71oGZNJGUr0w   Billy            16 2012-02-04 00:03:23   
459639   pHSg4d5tbXlw7BJ6SvYfiQ  Olivia             2 2016-10-18 03:18:54   
286692   HtrAxjwD6UgWixZECM93gw    Tina             7 2015-05-30 02:13:46   
1835373  Mhqm-Vkwoycb9SXBjdCRuw  Nicole             7 2012-11-05 20:21:59   
1751446  KDuRgWryzks0XSF_6nBj9g   Zeina             1 2011-07-09 00:14:36   

         useful  funny  cool elite  \
2077614      11     10     6         
459639        0      0     0         
286692        9      5     4         
1835373       2      0     1         
1751446       0      0     1         

                                                   friends  fans  \
2077614  MlU81KLZrTZDpFv7z-T9dA, zkftpSdmLrXqKBf-z7VuUg...     1   
459639   3GiQ1K8pkUe8sAXzw6o3Lg, LnmtRIJ3K4sauCCibAyIKA...     0   
286692   0ThNE_BH2qCd3nsz2THN9g, tvSxw3OCwtKPenLXsiqhPA...     0   
1835373  olKHknXzSZ4GIGhP4HemeQ, IO_uT32zuydizKejjUymoQ...     1   
1751446                                               None     0   

         average_stars  compliment_hot  compliment_more  compliment_profile  \
2077614           3.50               2                2                   0   
459639            4.00               0                0                   0   
286692            3.88               0                0                   0   
1835373           5.00               0                0                   0   
1751446           2.00               0                0                   0   

         compliment_cute  compliment_list  compliment_note  compliment_plain  \
2077614                0                0                1                 0   
459639                 0                0                0                 0   
286692                 0                0                1                 1   
1835373                0                0                0                 0   
1751446                0                0                0                 0   

         compliment_cool  compliment_funny  compliment_writer  \
2077614                2                 2                  0   
459639                 0                 0                  0   
286692                 0                 0                  0   
1835373                0                 0                  0   
1751446                0                 0                  0   

         compliment_photos  
2077614                  0  
459639                   0  
286692                   0  
1835373                  0  
1751446                  0

# user - without complimnets

In [ ]:
df = pd.read_parquet('/content/drive/MyDrive/DataYelpGoogle/user.parquet', engine='fastparquet', columns=['user_id',	'name',	'review_count',	'yelping_since',	'useful',	'funny',	'cool',	'elite',	'friends',	'fans',  'average_stars'])

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2105597 entries, 0 to 2105596
Data columns (total 11 columns):
 #   Column         Dtype  
---  ------         -----  
 0   user_id        object 
 1   name           object 
 2   review_count   int64  
 3   yelping_since  object 
 4   useful         int64  
 5   funny          int64  
 6   cool           int64  
 7   elite          object 
 8   friends        object 
 9   fans           int64  
 10  average_stars  float64
dtypes: float64(1), int64(5), object(5)
memory usage: 176.7+ MB


In [ ]:
# cambiar tipo de datos
df.review_count = df.review_count.astype('int32')
df.yelping_since = pd.to_datetime(df['yelping_since'], format='%Y-%m-%d')
df.useful = df.useful.astype('int32')
df.funny = df.funny.astype('int32')
df.cool = df.cool.astype('int32')
df.fans = df.fans.astype('int32')
df.average_stars = df.average_stars.astype('float32')

In [ ]:
# show data
df.sample(5)

user_id     name  review_count       yelping_since  \
51514    53KGxCpaTcyKF9hu0x4D_A        Y             3 2015-12-19 14:18:58   
562364   0T_wOCvEFWQJ3fsyh2U_0Q      Kim             1 2017-09-03 01:04:18   
494622   -WWO4_gyM0qG4OA8bkgnEA  Mariana            16 2014-06-14 23:32:19   
1290685  gHaBI95lNM11zNcOmGv8VQ      Kit             6 2017-08-01 00:19:40   
676327   Z2xS4C4RfW1XInMozKCQIg      Jon             2 2012-12-30 22:44:06   

         useful  funny  cool elite  \
51514         0      0     0         
562364        0      0     0         
494622        8      5     2         
1290685       1      0     0         
676327        1      0     0         

                                                   friends  fans  \
51514    5g-zWaEirwpZHB5g9kBoKg, L5p3DOSujoGsDaHzTBAFNg...     0   
562364                                                None     0   
494622                              IcsPT0B77NTfSRmm5NoIRw     0   
1290685  iuAbhhRR96OmnF4GxwGZ-Q, ZvQvOYezXfKnZ5--RlU6qA...     0   
676327   -Jq5yZ50CfDEFcnBtdT-tQ, MLttmBFV1rnoj5SdotedYg...     0   

         average_stars  
51514             3.67  
562364            5.00  
494622            3.50  
1290685           4.50  
676327            4.00

In [ ]:
# cantidad nulos
df.isna().sum()

user_id          0
name             0
review_count     0
yelping_since    0
useful           0
funny            0
cool             0
elite            0
friends          0
fans             0
average_stars    0
dtype: int64

In [ ]:
# exploracion variable numericas
df.describe()

review_count        useful         funny          cool          fans  \
count  2.105597e+06  2.105597e+06  2.105597e+06  2.105597e+06  2.105597e+06   
mean   2.846875e+01  5.516853e+01  2.267667e+01  3.143013e+01  1.913746e+00   
std    1.042146e+02  7.388068e+02  4.628811e+02  6.379274e+02  2.436510e+01   
min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
25%    2.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
50%    6.000000e+00  3.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
75%    1.900000e+01  1.500000e+01  3.000000e+00  4.000000e+00  0.000000e+00   
max    1.747300e+04  2.062960e+05  1.858230e+05  1.998780e+05  1.249700e+04   

       average_stars  
count   2.105597e+06  
mean    3.641863e+00  
std     1.159519e+00  
min     1.000000e+00  
25%     3.000000e+00  
50%     3.880000e+00  
75%     4.500000e+00  
max     5.000000e+00

In [ ]:
# duplicados?
df[df.duplicated()]

user_id         name  review_count  \
1987897  qVc8ODYU5SZjKXVBgXdI7w       Walker           585   
1987898  j14WgRoU_-2ZE1aw1dXrJg       Daniel          4333   
1987899  2WnXYQFK0hXEoTxPtV2zvg        Steph           665   
1987900  SZDeASXq7o05mMNLshsdIA         Gwen           224   
1987901  hA5lMy-EnncsH4JoR-hFGQ        Karen            79   
...                         ...          ...           ...   
2105592  4QGxxakRZeOlg_qDuxmTeQ     Jennilee            38   
2105593  tmelBbVBGAzXBVfH2u_R6g        Gerry            19   
2105594  tpBznnD6uJN3m_pJubj09w        Emily            26   
2105595  Kst_srPw7GdYydMFYdCtzw  Heatheranne            25   
2105596  mZl-mH3ZIWLRSd_FfvrXFg      Jessica            30   

              yelping_since  useful  funny   cool  \
1987897 2007-01-25 16:47:26    7217   1259   5994   
1987898 2009-01-25 04:35:42   43091  13066  27281   
1987899 2008-07-25 10:41:00    2086   1010   1003   
1987900 2005-11-29 04:38:33     512    330    299   
1987901 2007-01-05 19:40:59      29     15      7   
...                     ...     ...    ...    ...   
2105592 2012-01-19 23:33:02      74      9      6   
2105593 2009-06-09 16:34:54      14      5      2   
2105594 2013-08-13 23:18:11       4      1      2   
2105595 2015-01-10 00:06:25      21      2      5   
2105596 2015-11-30 21:10:17      27      1     17   

                                                     elite  \
1987897                                               2007   
1987898  2009,2010,2011,2012,2013,2014,2015,2016,2017,2...   
1987899                           2009,2010,2011,2012,2013   
1987900                                     2009,2010,2011   
1987901                                                      
...                                                    ...   
2105592                                                      
2105593                                                      
2105594                                                      
2105595                                                      
2105596                                                      

                                                   friends  fans  \
1987897  NSCy54eWehBJyZdG2iE84w, pe42u7DcCH2QmI81NX-8qA...   267   
1987898  ueRPE0CX75ePGMqOFVj6IQ, 52oH4DrRvzzl8wh5UXyU0A...  3138   
1987899  LuO3Bn4f3rlhyHIaNfTlnA, j9B4XdHUhDfTKVecyWQgyA...    52   
1987900  enx1vVPnfdNUdPho6PH_wg, 4wOcvMLtU6a9Lslggq74Vg...    28   
1987901  PBK4q9KEEBHhFvSXCUirIw, 3FWPpM7KU1gXeOM_ZbYMbA...     1   
...                                                    ...   ...   
2105592                             kmwNG5LZSHFmveg6wYYdrw     0   
2105593  BFYdCAMFyjYHDwesndEXEg, _9fTIqfSJc7g3V_o76XRVg...     1   
2105594  bKV3ly2MuK-K1cptMrFknQ, liel18zRoSB4tEkUP7i6Cg...     0   
2105595  dzHTk52vbGtbktRm_B-wEg, fOfFLV7IbBDN6lzARaLqdg...     0   
2105596  pou3BbKsIozfH50rxmnMew, JT3rDC7-hprCSObnihMeHw...     3   

         average_stars  
1987897           3.91  
1987898           3.74  
1987899           3.32  
1987900           4.27  
1987901           3.54  
...                ...  
2105592           2.98  
2105593           3.68  
2105594           3.85  
2105595           2.85  
2105596           4.13  

[117700 rows x 11 columns]

In [ ]:
# elimina duplicados
df.drop_duplicates(inplace=True)
df.reset_index(drop=True, inplace=True) # reset index

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1987897 entries, 0 to 1987896
Data columns (total 11 columns):
 #   Column         Dtype         
---  ------         -----         
 0   user_id        object        
 1   name           object        
 2   review_count   int32         
 3   yelping_since  datetime64[ns]
 4   useful         int32         
 5   funny          int32         
 6   cool           int32         
 7   elite          object        
 8   friends        object        
 9   fans           int32         
 10  average_stars  float32       
dtypes: datetime64[ns](1), float32(1), int32(5), object(4)
memory usage: 121.3+ MB


In [ ]:
# cantidad de usuarios unicos
df['user_id'].nunique()

1987897

In [ ]:
# cantidad de usuarios sin reviews
df[df['review_count']==0]

user_id       name  review_count       yelping_since  \
81766    1zEwvAm5TqJh7eqYShNVQA        Amy             0 2012-12-29 20:14:47   
146673   T8qn_FUg24LsdPaax4plNA      Kevin             0 2012-09-15 06:28:28   
332202   Pflag9DDmtagR6a-2Z8Jmw  Elizabeth             0 2013-12-26 19:48:37   
344992   MfQc-rOY5Qu_uRcoXDNzKA   Michelle             0 2021-04-09 17:06:53   
346327   OSSYJJIGVr53Tk61ZycITQ      Laura             0 2021-06-27 22:50:46   
388017   SfhWhnf-TuTjEulNwhVasA        Meg             0 2019-10-29 21:18:02   
449813   7IBB8V7G-2d4pWdxySAgew     Lyssie             0 2016-02-21 22:59:21   
510909   h2xeEEOpFyZHBnezJy1pYQ    Gemmika             0 2021-12-07 23:30:55   
543851   889Zs1Xyz6kucIShNdCJjg       Dawn             0 2022-01-18 05:42:13   
667520   ognrtkoN_v890L70KtnkcQ        Bre             0 2019-07-10 14:59:44   
692633   YHceEmeft2GZXpGrVApXQQ      Bella             0 2020-03-12 20:54:48   
729570   STCJB07I94MN34hOzovcZQ    Jazmine             0 2018-06-17 18:44:48   
807508   q7H2tz-o2Zjnl8mJHxuL6g   Chandler             0 2018-10-25 14:21:47   
814758   AmolTw60VjHqfEAHfjgVDQ      Alexa             0 2018-09-21 22:40:54   
848960   LXULQLXqjxG6_B5OOJf-MA     Steven             0 2015-01-22 02:40:06   
858234   IcuZF8KioeG-8uXqR4QqPA       Mike             0 2018-09-11 23:24:15   
904878   cco5VSBNHfow4jfLgjZG8A       Ivan             0 2017-09-27 21:07:26   
945442   6Vxh445RzNEgBT6ZTBOGhg      Patti             0 2019-09-09 20:40:42   
955812   ih2hlbMfrdiJrUTrT48fVQ       Bill             0 2021-09-01 15:14:00   
1067208  MJ1ht_fX0A8s6IK9Bi4joA    Anthony             0 2019-05-20 22:30:02   
1083764  QnNdLfcg1wd6-hOuvZyfsQ      Sheep             0 2017-05-13 19:06:36   
1091912  DaAV2TKAKE0A6gSBOLHbng     Mariel             0 2020-12-13 03:56:09   
1103785  Es2YpKHEgg0JUIazTMbYVw    Jasmine             0 2016-06-26 13:06:14   
1115141  ytY6AUF84s1jIv8e7-cDYg       Jack             0 2016-12-01 23:21:24   
1129442  qXHJa8Tf7Ef_wieJLIatxw     Daniel             0 2020-07-15 18:26:55   
1144789  909VYxYlC3ZZMaeDcZWmzw    Michael             0 2021-12-04 02:55:06   
1165484  wPILFUtXRSMd0xeYOstioA     Jeanne             0 2022-01-19 17:15:47   
1181169  WsBewXHyJSW7o_m6ImhFMQ    Colette             0 2022-01-16 21:18:36   
1233430  VuhY3tqUjMsFoV5ERRT58g      Barry             0 2014-06-24 21:07:59   
1239615  XrdIKaBn1BNljeJKEYQs6Q     Sierra             0 2017-03-15 20:30:23   
1307220  BONgr-63UcgPHwXxNr8jpg       Gimo             0 2015-02-01 01:08:57   
1311629  FlFkI3oQMBus2cayl-KciA     Olivia             0 2021-06-14 23:32:05   
1353652  9vfD0kRZ3nNXAWfS1OAs_w       Jodi             0 2010-08-22 12:05:23   
1389210  U7jYgJFjDOQysdTZIhHqXw      Sarah             0 2019-08-14 20:40:26   
1415913  ZkzSQX4FDNmpL1IJLKWZ7g      Kayla             0 2017-05-19 20:01:12   
1451799  -KJSgjfIh3rjFEdtfJ0lCQ       Kyle             0 2017-11-07 21:55:31   
1526599  MvTxxHzwrWKiloCrgTySYw     Austin             0 2019-05-05 01:43:35   
1553134  ouidYs3R2lhoHeiYJBm8bg       Mike             0 2022-01-18 14:24:10   
1564725  gglRKAawIJc1vXFoqa_MTw      Patti             0 2021-10-15 16:36:17   
1576319  FM2BqReuftlY6vSdNmIvhw    Stephen             0 2022-01-19 17:10:38   
1592904  dg97VPdC6J9iMbFWoxfmVw       Nick             0 2011-05-07 17:34:35   
1722481  iJe8jGzWclFANKV1FFgjbw     Maggie             0 2020-12-04 20:34:54   
1731289  yqgvaUa0TJEjeZWcu07jiQ     Jeanne             0 2021-01-30 04:44:37   
1830480  IEkS_T0KS3_yiCYRvXd_0Q         Tj             0 2012-09-24 19:04:08   
1874716  cVxHQacf0NxmfyTeLFz-CQ       Cole             0 2021-02-02 23:21:49   
1875074  yY-hAZcuJJ4mnPjXClkCHw    Carolyn             0 2020-03-27 17:42:47   
1875293  grNJBR-Vhill3Cd4nepbyQ     Meisha             0 2020-12-17 16:02:02   
1916677  _zEEc8nF7jtmzi_m8CYHug      Rully             0 2021-05-02 01:10:37   
1948772  dAWMZcDTDODFon45JOwsQg   Stefanie             0 2021-12-28 00:12:21   
1954765  ieAE0UIacoLUHtS

inconsistencia, si no tienen reviews no deberia tener votos useful, funny, cool tampoco average_stars

In [ ]:
df[df['elite'].str.contains('2021')]

user_id       name  review_count       yelping_since  \
1        j14WgRoU_-2ZE1aw1dXrJg     Daniel          4333 2009-01-25 04:35:42   
18       QF1Kuhs8iwLWANNZxebTow  Catherine           607 2009-04-27 20:25:54   
29       4ZaqBJqt7laPPs8xfWvr6A       Nina           801 2008-08-16 22:43:21   
32       NIhcRW6DWvk1JQhDhXwgOQ        Lia          2288 2005-12-30 13:47:19   
61       lquc6IF6uGIeRomDLu9UnA        Ron          1044 2007-10-04 01:20:53   
...                         ...        ...           ...                 ...   
1967716  RPvhoWzP4FTgv4fCOW_HZg  Stephanie            46 2016-04-28 02:09:44   
1971939  vS_6QTugCkLV1BVMYeA3hA       Paul           179 2010-11-21 13:34:52   
1976641  8NLJOmPqzOqLjfVle4Mwpg      Lynda           130 2010-02-06 08:04:28   
1981007  chQGOGeXRciXRjzbiiWcjQ    Armando            50 2017-09-09 14:32:37   
1981510  uRubSdDXddRccyNku9Vl8w    Michele            24 2021-03-16 04:34:31   

         useful  funny   cool  \
1         43091  13066  27281   
18         4573   3714   4149   
29         1944    616    737   
32        12773   4199   7971   
61         6025   1827   3151   
...         ...    ...    ...   
1967716      27      8     12   
1971939     249     50     39   
1976641     123     25     24   
1981007      31      0     10   
1981510      12      1      7   

                                                     elite  \
1        2009,2010,2011,2012,2013,2014,2015,2016,2017,2...   
18       2010,2011,2012,2013,2014,2015,2016,2017,2018,2...   
29       2010,2011,2012,2013,2014,2015,2016,2017,2018,2...   
32       2006,2007,2008,2009,2010,2011,2012,2013,2014,2...   
61       2010,2011,2012,2013,2014,2015,2016,2017,2018,2...   
...                                                    ...   
1967716                                               2021   
1971939                                    2019,20,20,2021   
1976641                                         20,20,2021   
1981007                                               2021   
1981510                                               2021   

                                                   friends  fans  \
1        ueRPE0CX75ePGMqOFVj6IQ, 52oH4DrRvzzl8wh5UXyU0A...  3138   
18       dLts9bY66tXEFqYG03YFgw, SYDGMC7d5NnMiT1l_WQq5A...   131   
29       v56_o8iCNYGB2JFVIhOwQA, yFLNHDlCriteboCVKjtpbA...    75   
32       T1upaPMzuW7pNj74fO1rjA, CP28puvAEimt4ziuGTDaHA...   345   
61       E9lEZUyOMbTsN7IJl6GBJA, 3yxW_9twke2IIJySDR4_4Q...   241   
...                                                    ...   ...   
1967716                                               None     1   
1971939                                               None     1   
1976641                             VPT90UqjKUCeEA4YT1TH6Q     2   
1981007                                               None     0   
1981510                                               None     0   

         average_stars  
1                 3.74  
18                4.11  
29                3.41  
32                3.69  
61                3.90  
...                ...  
1967716           4.30  
1971939           3.39  
1976641           3.56  
1981007           4.10  
1981510           4.12  

[44542 rows x 11 columns]

In [ ]:
df[df['elite'].str.contains('2022')]

Empty DataFrame
Columns: [user_id, name, review_count, yelping_since, useful, funny, cool, elite, friends, fans, average_stars]
Index: []

en el dataset aun no se registra usuarios elite para el año 2022. Los usuarios elite realizan mejores reviews (Thoughtful reviews, Awesome photos, Sending compliments, Up-voting reviews), para adquirrir el status elite un usuario se debe postular y esperar aprobacion (Use your real name on your profile, Add a real photo of yourself, Nominate yourself and tell us why you want in).  

In [ ]:
# usuarios con mayor numero de reviews
df.sort_values(by='review_count', ascending=False).head(5)

user_id    name  review_count       yelping_since  \
17169   Hi10sGSZNxQH3NLyWSZ1oA     Fox         17473 2009-05-26 11:33:58   
795339  8k3aO-mPeyhbR5HUucA5aA  Victor         16978 2007-12-08 14:56:45   
4723    hWDybu_KvYLSdEFzGrniTw   Bruce         16567 2009-03-08 21:47:44   
606393  RtGqdDBvvBCjcu5dUqwfzA   Shila         12868 2010-10-17 06:35:06   
152     P5bUL3Engv-2z6kKohB6qQ     Kim          9941 2006-05-31 21:27:42   

        useful   funny    cool  \
17169   206296  185823  195814   
795339  152979   84992  125516   
4723    173089   98459  144849   
606393   11109    3747    3292   
152      27549   10772   21375   

                                                    elite  \
17169            2014,2015,2016,2017,2018,2019,20,20,2021   
795339  2008,2009,2010,2011,2012,2013,2014,2015,2016,2...   
4723    2010,2011,2012,2013,2014,2015,2016,2017,2018,2...   
606393  2012,2013,2014,2015,2016,2017,2018,2019,20,20,...   
152     2006,2007,2008,2009,2010,2011,2012,2013,2014,2...   

                                                  friends  fans  average_stars  
17169   uZcWE40gWJHjLT_VyNBp1Q, N52nbyDKZte4PUfJVNpkGQ...  3493           3.77  
795339  Isb9ZKTK4L5nkOpyXypDPQ, d3rfsXGWtS7fpGsYfy_C7Q...  1462           3.35  
4723    sZ0Qxjn12H8jjHBPLQW32w, Lkk3DqzhvqwpiQKwwAyZcw...   867           3.67  
606393  hz79Mq7EGhHXZkIspAQtrg, LJE8g5sbnmu5Ua9YGN9SKw...   300           3.87  
152     hfY2Soz8LWSA0oRVcYuRmg, Y6_gZa4jVXv0d4qiMbbycw...   825           3.81

In [ ]:
# usuarios con mayor cantidad de useful reviews
df.sort_values(by='useful', ascending=False).head(5)

user_id     name  review_count       yelping_since  \
17169   Hi10sGSZNxQH3NLyWSZ1oA      Fox         17473 2009-05-26 11:33:58   
207385  --2vR0DIsmQ6WfcSzKWigw   Harald          1534 2012-11-27 14:19:33   
200787  JjXuiru1_ONzDkYVrHN0aw  Richard          1424 2009-08-23 13:03:27   
223840  lvthTfCQGD0qaEk6jCdRdQ    Ariel          2960 2010-07-28 16:51:07   
4723    hWDybu_KvYLSdEFzGrniTw    Bruce         16567 2009-03-08 21:47:44   

        useful   funny    cool  \
17169   206296  185823  195814   
207385  205765  173096  199878   
200787  183512  129638  173423   
223840  182788   54132  162667   
4723    173089   98459  144849   

                                                    elite  \
17169            2014,2015,2016,2017,2018,2019,20,20,2021   
207385      2013,2014,2015,2016,2017,2018,2019,20,20,2021   
200787  2011,2012,2013,2014,2015,2016,2017,2018,2019,2...   
223840                          2017,2018,2019,20,20,2021   
4723    2010,2011,2012,2013,2014,2015,2016,2017,2018,2...   

                                                  friends  fans  average_stars  
17169   uZcWE40gWJHjLT_VyNBp1Q, N52nbyDKZte4PUfJVNpkGQ...  3493           3.77  
207385  I0OmbiekW2dhHVczVylovA, pCU0h61iUKX_kXVc9CWO8A...   880           4.18  
200787  7Ie3f_4upzsMNAY07d19kA, SmkjLDoFYRBIfY0D8KevSQ...  3243           4.18  
223840  KnaNwxcn5j5-aaBVYPezAg, NZKgAXB3weu9f_DQbiJY7A...   808           4.47  
4723    sZ0Qxjn12H8jjHBPLQW32w, Lkk3DqzhvqwpiQKwwAyZcw...   867           3.67

In [ ]:
# usuarios con mayor cantidad de funny reviews
df.sort_values(by='funny', ascending=False).head(5)

user_id     name  review_count       yelping_since  \
17169   Hi10sGSZNxQH3NLyWSZ1oA      Fox         17473 2009-05-26 11:33:58   
207385  --2vR0DIsmQ6WfcSzKWigw   Harald          1534 2012-11-27 14:19:33   
200787  JjXuiru1_ONzDkYVrHN0aw  Richard          1424 2009-08-23 13:03:27   
800350  ax7SnXOTIpatbsmqHLqVow   Rohlin           972 2010-01-11 01:50:32   
4723    hWDybu_KvYLSdEFzGrniTw    Bruce         16567 2009-03-08 21:47:44   

        useful   funny    cool  \
17169   206296  185823  195814   
207385  205765  173096  199878   
200787  183512  129638  173423   
800350  112557  103618  107338   
4723    173089   98459  144849   

                                                    elite  \
17169            2014,2015,2016,2017,2018,2019,20,20,2021   
207385      2013,2014,2015,2016,2017,2018,2019,20,20,2021   
200787  2011,2012,2013,2014,2015,2016,2017,2018,2019,2...   
800350            2012,2013,2014,2015,2016,2017,2018,2019   
4723    2010,2011,2012,2013,2014,2015,2016,2017,2018,2...   

                                                  friends  fans  average_stars  
17169   uZcWE40gWJHjLT_VyNBp1Q, N52nbyDKZte4PUfJVNpkGQ...  3493           3.77  
207385  I0OmbiekW2dhHVczVylovA, pCU0h61iUKX_kXVc9CWO8A...   880           4.18  
200787  7Ie3f_4upzsMNAY07d19kA, SmkjLDoFYRBIfY0D8KevSQ...  3243           4.18  
800350  QE-aL9xrz1pc6MOY_rky6g, T1vkPJFoaOLCHFOMX3R__Q...   605           3.58  
4723    sZ0Qxjn12H8jjHBPLQW32w, Lkk3DqzhvqwpiQKwwAyZcw...   867           3.67

In [ ]:
# usuarios con mayor cantidad de cool reviews
df.sort_values(by='cool', ascending=False).head(5)

user_id     name  review_count       yelping_since  \
207385  --2vR0DIsmQ6WfcSzKWigw   Harald          1534 2012-11-27 14:19:33   
17169   Hi10sGSZNxQH3NLyWSZ1oA      Fox         17473 2009-05-26 11:33:58   
200787  JjXuiru1_ONzDkYVrHN0aw  Richard          1424 2009-08-23 13:03:27   
223840  lvthTfCQGD0qaEk6jCdRdQ    Ariel          2960 2010-07-28 16:51:07   
4723    hWDybu_KvYLSdEFzGrniTw    Bruce         16567 2009-03-08 21:47:44   

        useful   funny    cool  \
207385  205765  173096  199878   
17169   206296  185823  195814   
200787  183512  129638  173423   
223840  182788   54132  162667   
4723    173089   98459  144849   

                                                    elite  \
207385      2013,2014,2015,2016,2017,2018,2019,20,20,2021   
17169            2014,2015,2016,2017,2018,2019,20,20,2021   
200787  2011,2012,2013,2014,2015,2016,2017,2018,2019,2...   
223840                          2017,2018,2019,20,20,2021   
4723    2010,2011,2012,2013,2014,2015,2016,2017,2018,2...   

                                                  friends  fans  average_stars  
207385  I0OmbiekW2dhHVczVylovA, pCU0h61iUKX_kXVc9CWO8A...   880           4.18  
17169   uZcWE40gWJHjLT_VyNBp1Q, N52nbyDKZte4PUfJVNpkGQ...  3493           3.77  
200787  7Ie3f_4upzsMNAY07d19kA, SmkjLDoFYRBIfY0D8KevSQ...  3243           4.18  
223840  KnaNwxcn5j5-aaBVYPezAg, NZKgAXB3weu9f_DQbiJY7A...   808           4.47  
4723    sZ0Qxjn12H8jjHBPLQW32w, Lkk3DqzhvqwpiQKwwAyZcw...   867           3.67

In [ ]:
# usuarios con mayor numero de fans
df.sort_values(by='fans', ascending=False).head(5)

user_id     name  review_count       yelping_since  \
5444    37cpUoM8hlkSQfReIEBd-Q     Mike          1882 2009-04-25 14:54:26   
198889  hizGc5W1tBHPghM5YKCAtg    Katie          1825 2009-05-22 20:19:55   
17169   Hi10sGSZNxQH3NLyWSZ1oA      Fox         17473 2009-05-26 11:33:58   
200787  JjXuiru1_ONzDkYVrHN0aw  Richard          1424 2009-08-23 13:03:27   
1       j14WgRoU_-2ZE1aw1dXrJg   Daniel          4333 2009-01-25 04:35:42   

        useful   funny    cool  \
5444     22860   10965   15350   
198889   48711   32078   39747   
17169   206296  185823  195814   
200787  183512  129638  173423   
1        43091   13066   27281   

                                                    elite  \
5444    2012,2013,2014,2015,2016,2017,2018,2019,20,20,...   
198889  2009,2010,2011,2012,2013,2014,2015,2016,2017,2...   
17169            2014,2015,2016,2017,2018,2019,20,20,2021   
200787  2011,2012,2013,2014,2015,2016,2017,2018,2019,2...   
1       2009,2010,2011,2012,2013,2014,2015,2016,2017,2...   

                                                  friends   fans  \
5444    fLRfYMFIRzjX28OFTsCu4g, pEIzH_tNwK7pcqTViKm-RQ...  12497   
198889  wHrIy02OYTKYT1YdxPjhwg, 9myTU-JiWk-WaC9LlzK1iw...   3642   
17169   uZcWE40gWJHjLT_VyNBp1Q, N52nbyDKZte4PUfJVNpkGQ...   3493   
200787  7Ie3f_4upzsMNAY07d19kA, SmkjLDoFYRBIfY0D8KevSQ...   3243   
1       ueRPE0CX75ePGMqOFVj6IQ, 52oH4DrRvzzl8wh5UXyU0A...   3138   

        average_stars  
5444             4.39  
198889           4.23  
17169            3.77  
200787           4.18  
1                3.74

usuarios con gran cantidad de fans o alta cantidad de reviews podrian ser considerados influencers. las valoraciones por lo general son mas altas al promedio para el caso de los ususarios con gran cantidad de fans. A considerar, los influencers mueven masas y una review puede ser mas valiosa pensando en el marketing del negocio a pesar que no puede ser muy precisa en cuanto a objetividad

In [ ]:
# explorar usuarios calificacion promedio 5 estrellas
df.sort_values(by='average_stars', ascending=False).head(5)

user_id    name  review_count       yelping_since  \
1987896  wL5jPrLRVCK_Pmo4lM1zpA     Isa             2 2020-12-19 02:32:39   
505601   4DFQ0mxC5YsQvgxGIbOsNw   Sarah             4 2019-10-27 18:45:28   
505680   Zk2fA-VDhSaYPKUavDko_A  Dwight             6 2016-11-25 17:46:57   
1464692  LmXJ5Y_NjCzofvl5dq4a9g   Cindy             1 2015-10-31 03:50:51   
505678   wkO0Hc0si5piMBGWGn6m1A     Sue             4 2013-08-17 17:02:53   

         useful  funny  cool elite  \
1987896       0      0     0         
505601        1      1     0         
505680        1      0     0         
1464692       0      0     0         
505678        2      0     0         

                                                   friends  fans  \
1987896                                               None     0   
505601   fboaouc9DPpQy-1W5Mm31w, t3a5WjMpEkvI3FrOLBV7zg...     0   
505680   DzGCb2HkSxwDCgpwLo8HAA, xby4BQgR9YaJDJ77qYQBgw...     0   
1464692  -ByNItuf5aSCieakE7Z9VA, qqqQrpoz39_5PR73JN-ElA...     0   
505678   -dUTDM35-sPdkdsp3VKYeg, _WZDh17f-_EXHfX1YlVZQA...     0   

         average_stars  
1987896            5.0  
505601             5.0  
505680             5.0  
1464692            5.0  
505678             5.0

In [ ]:
# explorar usuarios calificacion promedio 1 estrella
df.sort_values(by='average_stars', ascending=True).head(5)

user_id    name  review_count       yelping_since  \
993948   0zSJgpvAs8Ogd03Ob26u6w   Brett             2 2016-08-20 20:58:51   
1517953  GaJpypDf0MVXH3NvbICE0Q   Linda             1 2019-05-06 15:58:32   
1517957  tmb--UbHualk0_thqE8QTQ    Matt             1 2021-06-04 16:44:44   
1517962  -wWqrx-orpv1Xv8sjcYXpQ   donna             1 2010-08-16 18:21:40   
1517969  OYlwgWXUjw-XknY_NbqNsA  Ashley             1 2013-01-13 17:07:07   

         useful  funny  cool elite friends  fans  average_stars  
993948        2      0     0          None     0            1.0  
1517953       0      0     0          None     0            1.0  
1517957       0      0     0          None     0            1.0  
1517962       2      0     0          None     0            1.0  
1517969       2      0     0          None     0            1.0

usuarios con calificaciones de 1 o 5 estrellas no han realizado muchas reviews, no poseen status elite, no tienen fans. Usuarios con review 1 estrella no tienen amigos

---------------

# business

-------------------

In [ ]:
df = pd.read_pickle("/content/drive/MyDrive/DataYelpGoogle/business.pkl")

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 150346 entries, 0 to 150345
Data columns (total 28 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   business_id   150346 non-null  object
 1   name          150346 non-null  object
 2   address       150346 non-null  object
 3   city          150346 non-null  object
 4   state         150343 non-null  object
 5   postal_code   150346 non-null  object
 6   latitude      150346 non-null  object
 7   longitude     150346 non-null  object
 8   stars         150346 non-null  object
 9   review_count  150346 non-null  object
 10  is_open       150346 non-null  object
 11  attributes    136602 non-null  object
 12  categories    150243 non-null  object
 13  hours         127123 non-null  object
 14  business_id   5 non-null       object
 15  name          5 non-null       object
 16  address       5 non-null       object
 17  city          5 non-null       object
 18  state         5 non-null

In [ ]:
# show data
df.sample(5)

business_id                       name  \
20451   x3G4YO4HgrPU-UeWQ_KxNA                  Mama Fu's   
119905  tOw7k1EIrrH5wCuvIGwTWw  Greyhound Package Express   
42728   3ONID-FiN6arsuNUjRz-1Q          Cjs Tire and Auto   
91250   FLaccoaJv7KdKdluX0u3Xg    Mend Me Massage Therapy   
115537  pqseW6lN4AAOrrxZuA2oYw          The Dance Network   

                                             address           city state  \
20451                2689 Gulf to Bay Blvd, Ste 1860     Clearwater    NJ   
119905                           1001 Filbert Street   Philadelphia    AB   
42728                                6801 Oldyork Rd   Philadelphia    IL   
91250   364 Wilmington W Chester Pike, Unit 7 Bldg B     Glen Mills    FL   
115537                            5130 Hollister Ave  Santa Barbara    FL   

       postal_code   latitude   longitude stars review_count is_open  \
20451        33759  27.960089  -82.727512   3.0           16       0   
119905       19107  39.952807  -75.157001   2.0            8       1   
42728        19126   40.05678   -75.13956   3.5           14       1   
91250        19342  39.858299   -75.54356   4.5            9       1   
115537       93111  34.435373 -119.802905   5.0            9       1   

                                               attributes  \
20451   {'GoodForKids': 'True', 'RestaurantsGoodForGro...   
119905                                               None   
42728              {'BusinessAcceptsCreditCards': 'True'}   
91250   {'AcceptsInsurance': 'False', 'WheelchairAcces...   
115537  {'GoodForKids': 'True', 'BusinessParking': 'No...   

                                               categories  \
20451                           Asian Fusion, Restaurants   
119905  Shipping Centers, Couriers & Delivery Services...   
42728   Oil Change Stations, Auto Repair, Tires, Autom...   
91250   Beauty & Spas, Physical Therapy, Tui Na, Tradi...   
115537  Fitness & Instruction, Education, Dance Studio...   

                                                    hours business_id name  \
20451                                                None         NaN  NaN   
119905  {'Monday': '5:30-23:59', 'Tuesday': '5:30-23:5...         NaN  NaN   
42728   {'Monday': '7:0-18:0', 'Tuesday': '7:0-18:0', ...         NaN  NaN   
91250   {'Monday': '9:30-21:30', 'Tuesday': '9:30-21:3...         NaN  NaN   
115537  {'Monday': '0:0-0:0', 'Tuesday': '12:0-20:0', ...         NaN  NaN   

       address city state postal_code latitude longitude stars review_count  \
20451      NaN  NaN   NaN         NaN      NaN       NaN   NaN          NaN   
119905     NaN  NaN   NaN         NaN      NaN       NaN   NaN          NaN   
42728      NaN  NaN   NaN         NaN      NaN       NaN   NaN          NaN   
91250      NaN  NaN   NaN         NaN      NaN       NaN   NaN          NaN   
115537     NaN  NaN   NaN         NaN      NaN       NaN   NaN          NaN   

       is_open attributes categories hours  
20451      NaN        NaN        NaN   NaN  
119905     NaN        NaN        NaN   NaN  
42728      NaN        NaN        NaN   NaN  
91250      NaN        NaN        NaN   NaN  
115537     NaN        NaN        NaN   NaN

probelma al leer archivo .pkl las columnas se duplican, por lo cual se hara un nuevo archivo .parquet en el cual se tendra en cuenta las priemras columnas y se tomara 5 registros validos de las demas

In [ ]:
# df1  columna 0 a columna 13
df1 = df.iloc[:, 0:14]

In [ ]:
# df2 columnas 14 en adelante
df2 = df.iloc[:, 14:28]

In [ ]:
# extraer registros validos de las columnas a descartar
df2 = df2[df2['business_id'].notnull()]

In [ ]:
# armar un nuevo dataframe
df = pd.concat([df1, df2], axis=0)

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 150351 entries, 0 to 1543
Data columns (total 14 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   business_id   150351 non-null  object
 1   name          150351 non-null  object
 2   address       150351 non-null  object
 3   city          150351 non-null  object
 4   state         150348 non-null  object
 5   postal_code   150351 non-null  object
 6   latitude      150351 non-null  object
 7   longitude     150351 non-null  object
 8   stars         150351 non-null  object
 9   review_count  150351 non-null  object
 10  is_open       150351 non-null  object
 11  attributes    136607 non-null  object
 12  categories    150248 non-null  object
 13  hours         127128 non-null  object
dtypes: object(14)
memory usage: 17.2+ MB


In [ ]:
# cambiar tipo de datos
df.latitude = df.latitude.astype('float64')
df.longitude = df.longitude.astype('float64')
df.stars = df.stars.astype('float32')
df.review_count = df.review_count.astype('int32')

In [ ]:
# show data
df.sample(5)

business_id                               name  \
88841   lOGzA2ABSDEsPfJ0P-ctNg                      Audio Express   
62239   p7dXsOE8BX9lBcDiA3N-ng                      All Hours Air   
37826   QHa9ifszrZCIqF8cuXpBYA           Asarco Mineral Discovery   
31937   sHd4awXmmBEyeqc38Sw7hg                         Mister BBQ   
131077  MBxcC2spB4juXq9lqExFKA  True Nails and Eyebrows Threading   

                            address       city state postal_code   latitude  \
88841              5004 N Oracle Rd     Tucson    TN       85704  32.297808   
62239   395 Freeport Blvd, Suite 12     Sparks    FL       89431  39.524999   
37826           1421 W Pima Mine Rd  Sahuarita    AZ       85629  31.997906   
31937            5163 N Glenwood St      Boise    CA       83714  43.651861   
131077     7010 Sheldon Rd, Ste 700      Tampa    MO       33615  28.014671   

         longitude  stars  review_count is_open  \
88841  -110.977176    2.5            51       1   
62239  -119.747535    4.0           120       1   
37826  -110.994900    4.5            12       1   
31937  -116.280732    4.5            47       0   
131077  -82.583739    3.0            19       1   

                                               attributes  \
88841   {'BusinessAcceptsCreditCards': 'True', 'Restau...   
62239   {'ByAppointmentOnly': 'True', 'BusinessAccepts...   
37826   {'GoodForKids': 'True', 'WiFi': 'u'no'', 'ByAp...   
31937   {'BYOB': 'False', 'BusinessAcceptsCreditCards'...   
131077  {'ByAppointmentOnly': 'True', 'BusinessAccepts...   

                                               categories  \
88841   Auto Parts & Supplies, Shopping, Electronics, ...   
62239   Heating & Air Conditioning/HVAC, Home Services...   
37826   Museums, Tours, Arts & Entertainment, Hotels &...   
31937   American (New), Restaurants, Barbeque, America...   
131077  Threading Services, Nail Salons, Beauty & Spas...   

                                                    hours  
88841   {'Monday': '0:0-0:0', 'Tuesday': '9:0-18:0', '...  
62239   {'Monday': '0:0-0:0', 'Tuesday': '0:0-0:0', 'W...  
37826   {'Tuesday': '9:0-15:0', 'Wednesday': '9:0-15:0...  
31937   {'Tuesday': '11:0-21:0', 'Wednesday': '11:0-21...  
131077  {'Monday': '9:30-19:0', 'Tuesday': '9:30-19:0'...

In [ ]:
# crear parquet columnas validas
df.to_parquet('business.parquet', index=False)

# business parquet

In [ ]:
df = pd.read_parquet('/content/drive/MyDrive/DataYelpGoogle/business.parquet')

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150351 entries, 0 to 150350
Data columns (total 14 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   business_id   150351 non-null  object 
 1   name          150351 non-null  object 
 2   address       150351 non-null  object 
 3   city          150351 non-null  object 
 4   state         150348 non-null  object 
 5   postal_code   150351 non-null  object 
 6   latitude      150351 non-null  float64
 7   longitude     150351 non-null  float64
 8   stars         150351 non-null  float32
 9   review_count  150351 non-null  int32  
 10  is_open       150351 non-null  int64  
 11  attributes    136607 non-null  object 
 12  categories    150248 non-null  object 
 13  hours         127128 non-null  object 
dtypes: float32(1), float64(2), int32(1), int64(1), object(9)
memory usage: 14.9+ MB


In [ ]:
# cambiar tipo de datos
df.is_open = df.is_open.astype('object')

In [ ]:
# show data
df.sample(5)

business_id                                name  \
110178  iZwtERULHsk1SSlIlKxKTQ                       Hair By Haley   
88793   fHgEvmxEhF6ayrQF4HjBxA                Slater's Ding-a-Wing   
28115   HJhy636kx3WPwRsxkQ2Q4A                         Diggs Pizza   
54464   NwJ7tMLDwA3E3Cnq80YUWw                 Supreme Water Sales   
36192   Yp6CgKIDfdxUEsFM_Iziqw  Lady In White Housekeeping Service   

                          address           city state postal_code   latitude  \
110178  7689 S Virginia St, Ste D           Reno    FL       89511  39.456435   
88793                                       Reno    PA       89501  39.525400   
28115              4646 S Cole Rd          Boise    FL       83709  43.560545   
54464          531 Main St, Ste E  Safety Harbor    FL       34695  27.991294   
36192                                 Swedesboro    FL       08085  39.747614   

         longitude  stars  review_count is_open  \
110178 -119.781782    5.0             8       1   
88793  -119.813527    4.0            25       1   
28115  -116.273697    4.5            93       1   
54464   -82.692935    5.0             5       1   
36192   -75.310465    3.5             9       1   

                                               attributes  \
110178  {'AcceptsInsurance': None, 'AgesAllowed': None...   
88793   {'AcceptsInsurance': None, 'AgesAllowed': None...   
28115   {'AcceptsInsurance': None, 'AgesAllowed': None...   
54464   {'AcceptsInsurance': None, 'AgesAllowed': None...   
36192   {'AcceptsInsurance': None, 'AgesAllowed': None...   

                                               categories  \
110178  Hair Salons, Beauty & Spas, Kids Hair Salons, ...   
88793                   Food Trucks, Street Vendors, Food   
28115                                  Pizza, Restaurants   
54464   Irrigation, Hot Tub & Pool, Shopping, Water Pu...   
36192   Office Cleaning, Home Cleaning, Window Washing...   

                                                    hours  
110178  {'Friday': '10:0-19:0', 'Monday': None, 'Satur...  
88793                                                None  
28115   {'Friday': '12:0-21:0', 'Monday': '0:0-0:0', '...  
54464   {'Friday': '9:0-16:0', 'Monday': '9:0-16:0', '...  
36192   {'Friday': '6:0-23:0', 'Monday': '6:0-23:0', '...

In [ ]:
# cantidad nulos
df.isna().sum()

business_id         0
name                0
address             0
city                0
state               3
postal_code         0
latitude            0
longitude           0
stars               0
review_count        0
is_open             0
attributes      13744
categories        103
hours           23223
dtype: int64

In [ ]:
# exploracion variables numericas
df.describe()

latitude      longitude          stars   review_count
count  150351.000000  150351.000000  150351.000000  150351.000000
mean       36.671267     -89.357466       3.596717      44.865947
std         5.872883      14.918691       0.974411     121.118222
min        27.555127    -120.095137       1.000000       5.000000
25%        32.187303     -90.357818       3.000000       8.000000
50%        38.777436     -86.121178       3.500000      15.000000
75%        39.954037     -75.421512       4.500000      37.000000
max        53.679197     -73.200457       5.000000    7568.000000

In [ ]:
df['business_id'].nunique()

150346

In [ ]:
# duplicados?
df[df.duplicated(subset=['business_id',	'address',	'city',	'state',	'postal_code',	'latitude',	'longitude',	'stars',	'review_count',	'is_open'])]

business_id       name                             address  \
150346  6_Zf2QsRHCH1RhbzA9inyw  STarbucks               8602 Hillsborough Ave   
150347  Iipfa9Kaac9XH5Vo8GQwvg  STarbucks                     5956 153 Avenue   
150348  oY6osC4Ch3PvguA7D1hh_w  STarbucks                    1251 Route 130 S   
150349  pxuMxh5oC7W-PDN9wwvozg  STarbucks                     219 N Center St   
150350  lo2VbA7Lwzu3i23pQpmzlw  STarbucks  1125 S Black Horse Pike, Space 575   

                       city state postal_code   latitude   longitude  stars  \
150346                Tampa    LA       33615  27.996370  -82.580954    3.0   
150347             Edmonton    PA     T5Y 2W1  53.616147 -113.432424    3.5   
150348          Cinnaminson    AZ       08077  40.000157  -74.987313    4.5   
150349                 Reno    FL       89501  39.527772 -119.812518    3.0   
150350  Gloucester Township    NJ       08012  39.784030  -75.052070    3.0   

        review_count is_open  \
150346            56       1   
150347            10       1   
150348             7       1   
150349            44       0   
150350            15       1   

                                               attributes          categories  \
150346  {'AcceptsInsurance': None, 'AgesAllowed': None...  Coffee & Tea, Food   
150347  {'AcceptsInsurance': None, 'AgesAllowed': None...  Coffee & Tea, Food   
150348  {'AcceptsInsurance': None, 'AgesAllowed': None...  Coffee & Tea, Food   
150349  {'AcceptsInsurance': None, 'AgesAllowed': None...  Coffee & Tea, Food   
150350  {'AcceptsInsurance': None, 'AgesAllowed': None...  Food, Coffee & Tea   

                                                    hours  
150346  {'Friday': '5:30-21:0', 'Monday': '5:30-21:0',...  
150347  {'Friday': '5:30-22:0', 'Monday': '5:30-22:0',...  
150348  {'Friday': '5:30-20:0', 'Monday': '5:30-20:0',...  
150349  {'Friday': '5:30-20:0', 'Monday': '5:30-19:0',...  
150350  {'Friday': '9:30-20:0', 'Monday': '0:0-0:0', '...

In [ ]:
# elimina duplicados
df.drop_duplicates(subset=['business_id',	'address',	'city',	'state',	'postal_code',	'latitude',	'longitude',	'stars',	'review_count',	'is_open'], inplace=True)
df.reset_index(drop=True, inplace=True) # reset index

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150346 entries, 0 to 150345
Data columns (total 14 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   business_id   150346 non-null  object 
 1   name          150346 non-null  object 
 2   address       150346 non-null  object 
 3   city          150346 non-null  object 
 4   state         150343 non-null  object 
 5   postal_code   150346 non-null  object 
 6   latitude      150346 non-null  float64
 7   longitude     150346 non-null  float64
 8   stars         150346 non-null  float32
 9   review_count  150346 non-null  int32  
 10  is_open       150346 non-null  object 
 11  attributes    136602 non-null  object 
 12  categories    150243 non-null  object 
 13  hours         127123 non-null  object 
dtypes: float32(1), float64(2), int32(1), object(10)
memory usage: 14.9+ MB


In [ ]:
# verificar nulos state
df[df['state'].isna()]

business_id                      name  \
0  Pns2l4eNsfO8kk83dixA6A  Abby Rappoport, LAC, CMQ   
1  mpf3x-BjTdTEA3yCZrAYPw             The UPS Store   
2  tUFrWirKiKi_TAnsVWINQQ                    Target   

                           address           city state postal_code  \
0           1616 Chapala St, Ste 2  Santa Barbara  None       93101   
1  87 Grasso Plaza Shopping Center         Affton  None       63123   
2             5255 E Broadway Blvd         Tucson  None       85711   

    latitude   longitude  stars  review_count is_open  \
0  34.426679 -119.711197    5.0             7       0   
1  38.551126  -90.335695    3.0            15       1   
2  32.223236 -110.880452    3.5            22       0   

                                          attributes  \
0  {'AcceptsInsurance': None, 'AgesAllowed': None...   
1  {'AcceptsInsurance': None, 'AgesAllowed': None...   
2  {'AcceptsInsurance': None, 'AgesAllowed': None...   

                                          categories  \
0  Doctors, Traditional Chinese Medicine, Naturop...   
1  Shipping Centers, Local Services, Notaries, Ma...   
2  Department Stores, Shopping, Fashion, Home & G...   

                                               hours  
0                                               None  
1  {'Friday': '8:0-18:30', 'Monday': '0:0-0:0', '...  
2  {'Friday': '8:0-23:0', 'Monday': '8:0-22:0', '...

In [ ]:
# numero de estados
df['state'].unique()

array([None, 'CA', 'MO', 'AZ', 'PA', 'TN', 'FL', 'IN', 'LA', 'AB', 'NV',
       'ID', 'DE', 'IL', 'NJ', 'NC', 'CO', 'WA', 'HI', 'UT', 'TX', 'MT',
       'MI', 'SD', 'XMS', 'MA', 'VI', 'VT'], dtype=object)

In [ ]:
# cantidad de business por estado
df['state'].value_counts()

PA     34039
FL     26329
TN     12056
IN     11246
MO     10913
LA      9924
AZ      9912
NJ      8536
NV      7715
AB      5573
CA      5203
ID      4467
DE      2265
IL      2144
TX         4
CO         3
WA         2
HI         2
MA         2
NC         1
UT         1
MT         1
MI         1
SD         1
XMS        1
VI         1
VT         1
Name: state, dtype: int64

business de solo 27 estados y en algunos estados solo 1 business

In [ ]:
# calificacion promedio por estado
df.groupby(['state'])['stars'].mean().sort_values(ascending=False)

state
CO     5.000000
MA     4.750000
XMS    4.500000
VT     4.000000
UT     4.000000
MT     4.000000
HI     4.000000
MI     4.000000
MO     3.618116
NV     3.612314
AB     3.603535
LA     3.599657
FL     3.599605
ID     3.599060
IN     3.598702
PA     3.595875
AZ     3.589538
IL     3.589319
NJ     3.585227
CA     3.584086
TN     3.582117
DE     3.561589
WA     3.500000
TX     3.250000
NC     3.000000
SD     2.500000
VI     2.500000
Name: stars, dtype: float32

In [ ]:
# business con mayor cantidad de reviews
df.sort_values(by='review_count', ascending=False).head(5)

business_id                                name  \
113731  _ab50qdWOk0DdB6XOrBitw                   Acme Oyster House   
112552  ac1AeYqs8Z4_e2X5M3if2A                        Oceana Grill   
91757   GXFMD0Z4jEVZBCsbPf4CTQ  Hattie B’s Hot Chicken - Nashville   
143157  ytynqOUb3hjKeJfRj5Tshw             Reading Terminal Market   
147081  oBNrLz4EDhiscSlbOl8uAw          Ruby Slipper - New Orleans   

                 address          city state postal_code   latitude  \
113731  724 Iberville St   New Orleans    AZ       70130  29.954273   
112552      739 Conti St   New Orleans    FL       70130  29.956231   
91757     112 19th Ave S     Nashville    NJ       37203  36.151387   
143157      51 N 12th St  Philadelphia    PA       19107  39.953341   
147081   200 Magazine St   New Orleans    FL       70130  29.951025   

        longitude  stars  review_count is_open  \
113731 -90.068965    4.0          7568       1   
112552 -90.067563    4.0          7400       1   
91757  -86.796603    4.5          6093       1   
143157 -75.158855    4.5          5721       1   
147081 -90.067394    4.5          5193       1   

                                               attributes  \
113731  {'AcceptsInsurance': None, 'AgesAllowed': None...   
112552  {'AcceptsInsurance': None, 'AgesAllowed': None...   
91757   {'AcceptsInsurance': None, 'AgesAllowed': None...   
143157  {'AcceptsInsurance': None, 'AgesAllowed': None...   
147081  {'AcceptsInsurance': None, 'AgesAllowed': None...   

                                               categories  \
113731  Live/Raw Food, Seafood, Restaurants, Cajun/Creole   
112552  Restaurants, Seafood, Cajun/Creole, Breakfast ...   
91757   American (Traditional), Chicken Shop, Southern...   
143157  Candy Stores, Shopping, Department Stores, Fas...   
147081  Restaurants, American (Traditional), American ...   

                                                    hours  
113731  {'Friday': '11:0-22:0', 'Monday': '11:0-22:0',...  
112552  {'Friday': '8:0-1:0', 'Monday': '8:0-1:0', 'Sa...  
91757   {'Friday': '11:0-0:0', 'Monday': '0:0-0:0', 'S...  
143157  {'Friday': '8:0-18:0', 'Monday': '8:0-18:0', '...  
147081  {'Friday': '7:30-14:0', 'Monday': '0:0-0:0', '...

Fitrar Wine

In [ ]:
df.dropna(subset=['categories'], inplace=True)

In [ ]:
df[df['categories'].str.contains('Vine|Wine')]

business_id                            name  \
20      WKMJwqnfZKsAae75RMP6jA  Roast Coffeehouse and Wine Bar   
47      lk9IwjZXqUMqqOhM774DtQ                Caviar & Bananas   
81      7clCBgNbd-x2Wj96lZ6Mjw       Bier Brewery and Tap Room   
87      YNjyv0gfOr2g8lbmUpTnKg                     Copper Vine   
110     VrGxoSNZfSiT-G4OqRRGIw                Sunset 44 Bistro   
...                        ...                             ...   
150235  mzTTdvQGkUHZ8ii7OnZC5w                    Trader Joe's   
150254  Ld3qgGua7MqrFnZY8AIo0A                    Café Ventana   
150264  Bo-GGwQNcLVwyKeYkbfAYg                     Vino Divino   
150292  esBGrrmuZzSiECyRBoKvvA   Colony Grill - St. Petersburg   
150322  2MAQeAqmD8enCT2ZYqUgIQ     The Melting Pot - Nashville   

                     address            city state postal_code   latitude  \
20       10359 104 Street NW        Edmonton    LA     T5J 1B9  53.546045   
47             2031 Broadway       Nashville    PA       37203  36.148371   
81            5133 E 65th St    Indianapolis    AZ       46220  39.875532   
87           1001 Poydras St     New Orleans    PA       70112  29.950647   
110          118 W Adams Ave     Saint Louis    PA       63122  38.583223   
...                      ...             ...   ...         ...        ...   
150235        2121 Market St    Philadelphia    FL       19103  39.954178   
150254      3919 W Pine Blvd     Saint Louis    LA       63108  38.638030   
150264    2012 De La Vina St   Santa Barbara    AB       93105  34.429154   
150292       670 Central Ave  St. Petersburg    FL       33701  27.770872   
150322  166 2nd Ave N, Ste A       Nashville    PA       37201  36.163875   

         longitude  stars  review_count is_open  \
20     -113.499169    4.0            40       0   
47      -86.798895    3.5           159       0   
81      -86.082857    4.5           139       1   
87      -90.074427    4.5           350       1   
110     -90.407187    4.0            73       1   
...            ...    ...           ...     ...   
150235  -75.175980    4.0           494       1   
150254  -90.242133    3.5           290       0   
150264 -119.717222    5.0            19       1   
150292  -82.643069    4.5            38       1   
150322  -86.776311    4.0           204       0   

                                               attributes  \
20      {'AcceptsInsurance': None, 'AgesAllowed': None...   
47      {'AcceptsInsurance': None, 'AgesAllowed': None...   
81      {'AcceptsInsurance': None, 'AgesAllowed': None...   
87      {'AcceptsInsurance': None, 'AgesAllowed': None...   
110     {'AcceptsInsurance': None, 'AgesAllowed': None...   
...                                                   ...   
150235  {'AcceptsInsurance': None, 'AgesAllowed': None...   
150254  {'AcceptsInsurance': None, 'AgesAllowed': None...   
150264  {'AcceptsInsurance': None, 'AgesAllowed': None...   
150292  {'AcceptsInsurance': None, 'AgesAllowed': None...   
150322  {'AcceptsInsurance': None, 'AgesAllowed': None...   

                                               categories  \
20      Coffee & Tea, Food, Cafes, Bars, Wine Bars, Re...   
47      Coffee & Tea, Restaurants, Wine Bars, Bars, Ni...   
81                  Food, Beer, Wine & Spirits, Breweries   
87      Nightlife, Pubs, Event Planning & Services, Wi...   
110     Bars, Lounges, American (New), Restaurants, Ni...   
...                                                   ...   
150235  Beer, Wine & Spirits, Shopping, Grocery, Flori...   
150254  Food, Breakfast & Brunch, Cajun/Creole, Restau...   
150264  Event Planning & Services, Wine Tasting Room, ...   
150292  Bars, Beer Bar, Nightlife, Wine Bars, Pizza, R...   
150322    Fondue, Beer, Wine & Spirits, Food, Restaurants   

                                                    hours  
20      {'Friday': '8:0-18:0', 'Monday': '8:0-18:0', '...  
47      {'Friday': '7:0-17:0', 'Monday': '7:0-17:0', '...  
81      {'Friday': '13:0-21:0', 'Monday': 

In [ ]:
df = df[df['categories'].str.contains('Wine')]

In [ ]:
df = df[['business_id',	'name',	'address',	'city',	'state',	'postal_code',	'latitude',	'longitude',	'stars',	'review_count',	'categories']]

In [ ]:
df.sample(5)

business_id                                           name  \
62036   2_j80N3i1neQzf0BArnaGw                                     Save A Lot   
104984  raXrqHPI0MmUdpJ9PC_T1g                                     la Padella   
61176   9kFMEa74Y79PKFGVO7dIPQ                                     Vista Peru   
64241   aXWzxc2fYNBpr1xPLK2-qw  Fine Wine & Good Spirits - Premium Collection   
70096   ZeipYKyS88eRK2EDCycgFA                             R & R Liquor Store   

                            address          city state postal_code  \
62036   7351 Spring Hill Dr, Unit 1   Spring Hill    LA       34606   
104984                 100 State Rd         Media    FL       19063   
61176                   20 S 2nd St  Philadelphia    PA       19106   
64241        401 Franklin Mills Cir  Philadelphia    TN       19154   
70096             1043 Jefferson St     Nashville    FL       37208   

         latitude  longitude  stars  review_count  \
62036   28.452858 -82.595743    3.0             5   
104984  39.929514 -75.382169    4.0            50   
61176   39.949418 -75.143959    4.0           193   
64241   40.085472 -74.964806    4.0             5   
70096   36.171617 -86.796651    3.0             6   

                                               categories  
62036   Food, Grocery, Beer, Wine & Spirits, Discount ...  
104984  Cocktail Bars, Italian, Nightlife, Bars, Wine ...  
61176   Peruvian, Nightlife, Bars, Cocktail Bars, Wine...  
64241                          Beer, Wine & Spirits, Food  
70096                          Food, Beer, Wine & Spirits

In [ ]:
df.shape

(3655, 11)

In [ ]:
df['state'].value_counts()

PA    800
FL    650
IN    282
AZ    271
TN    267
MO    249
NJ    229
LA    221
NV    184
AB    153
ID    132
CA    111
DE     56
IL     49
HI      1
Name: state, dtype: int64

In [ ]:
df.to_parquet('Yelp_Business_Filter_Wine.parquet', index=False)

----------------

# tip

-----------------

In [ ]:
df = pd.read_parquet('/content/drive/MyDrive/DataYelpGoogle/tip.parquet', engine='fastparquet')

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 908915 entries, 0 to 908914
Data columns (total 5 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   user_id           908915 non-null  object        
 1   business_id       908915 non-null  object        
 2   text              908915 non-null  object        
 3   date              908915 non-null  datetime64[ns]
 4   compliment_count  908915 non-null  int64         
dtypes: datetime64[ns](1), int64(1), object(3)
memory usage: 34.7+ MB


In [ ]:
# show data
df.sample(5)

user_id             business_id  \
493368  YOXzvY44BRlp9eTUrUeNqA  _LgI5387qAmMuS8i6h69cg   
639560  pRkUTXOln1hz7_oS5DXyiA  LgM7qI5CmFAotFYcD_54dg   
267101  gru76BjkRY6UQVcw_yAxiQ  I3Uy20BKx7G-Fzz6hLIeWg   
699152  laQKsJ4PRq3cNmXhr9c1Nw  NLVsIjPW6ungGB4k1QenfQ   
205792  trf3Qcz8qvCDKXiTgjUcEg  8W03O-MpzfkYRgxuagrcPA   

                                                     text                date  \
493368  Must order whatever tres leches cake special t... 2019-02-27 21:49:23   
639560                                        Great food! 2019-06-13 23:33:53   
267101                    Gnocchi with crab is to DIE for 2011-01-14 02:33:56   
699152  I don't know what the bad tips are talking abo... 2013-05-15 01:35:04   
205792  Warning : No alfalfa sprouts (even though they... 2012-03-05 19:53:53   

        compliment_count  
493368                 0  
639560                 0  
267101                 0  
699152                 0  
205792                 0

In [ ]:
df['user_id'].nunique()

301758

In [ ]:
df['business_id'].nunique()

106193

In [ ]:
# crear columna year
df['year'] = df['date'].dt.year

In [ ]:
# filtrar año mayor igual 2017
df = df[df['year']>=2017]
df.reset_index(drop=True, inplace=True) # reset index

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 286942 entries, 0 to 286941
Data columns (total 6 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   user_id           286942 non-null  object        
 1   business_id       286942 non-null  object        
 2   text              286942 non-null  object        
 3   date              286942 non-null  datetime64[ns]
 4   compliment_count  286942 non-null  int64         
 5   year              286942 non-null  int64         
dtypes: datetime64[ns](1), int64(2), object(3)
memory usage: 13.1+ MB


In [ ]:
# show data filtro año 2017 - 2022
df.sample(5)

user_id             business_id  \
194457  0ZSSxlUujHpr76ZrcDKE4g  FEXhWNCMkv22qG04E83Qjg   
93980   ZjG9BzVyIpA4VlQ-pCxNfw  hLxk3KlKRNG6cCYjSfYmZA   
115380  EU6IjhPvOiHfIPtqVrYf4g  B9Z3i-n9y9FfMU3CQxtThA   
280321  ILiwonJxpH_T92nI6N6fxQ  s_4al4_I9YEpPguapUTSGw   
36229   nuxIF7lJ6IbS9A2mZvQHmA  eDqPBPit4nCy_qLEE6AA9A   

                                                     text                date  \
194457  Nice server in a busy place. Fresh hot donuts.... 2019-03-08 03:15:14   
93980   In hay-fever season it's a great place to visi... 2018-03-26 07:41:09   
115380                                            Amazing 2018-10-08 01:41:49   
280321  Nice gym, just gets crowded but otherwise good... 2017-04-03 22:43:34   
36229   This place is a stationery food truck is right... 2017-10-11 23:07:10   

        compliment_count  year  
194457                 0  2019  
93980                  0  2018  
115380                 0  2018  
280321                 0  2017  
36229                  0  2017

In [ ]:
# duplicados?
df[df.duplicated()]

user_id             business_id  \
10501   WQ8shYm0ghNDz97BuHI1fA  Sv1MEZP-mMfp8SmE0hwYEA   
13633   hGxVvXVg7IK4J2aTiGMSIQ  5RsVAkDnMrcSbErS6P1eew   
16776   3nBDXQT81w91zv8NPke8Mw  2JEzWTF9tazzjyzCTsOWQw   
26341   OrkHIX-BA2FVVG7v0VbeAg  TfVWvcKdpqBconG-fRzMyg   
34982   pJ8QYnUYb83Qn80aMOTjNQ  PVRs2-0uQYqfdhoCidwA_w   
38060   lajFU_Rs3kn7Gbt5REmKsg  kIiFXhJY5mymV8_25A7jrA   
39703   SNZJNz1WEIMECEk-TAYADQ  6Gx45ltNNZvjzVBO6ZgVOg   
64523   6ZwGm0WBtMgo5pCwyRDgdQ  diovHdArxKlT5ZgkTfDkRQ   
69495   1pdFXo6HykfxLE68cEE-ig  D9p7-HsY9llYP3BaCVg4DA   
83009   P1n1WG61-QM7gvnWJLc3fg  -QI8Qi8XWH3D8y8ethnajA   
137493  AzwLvR4wF6_c92AMHEQdTw  pWBhWaPdHLu09tYAIYIPNw   
149596  wCgOoPj2Dy7Fov6Bzh6oUQ  otEpgG6DdLiKtKtQyfM70g   
149902  hb6UOk4v5WQCHAg_T0EP1Q  r2IuwdvmEZycvudboTNg7A   
152067  uXzGKrWOhM8uB2sJ-D8Xew  R1DzXL1qjRyqwfDcl0L_RQ   
153167  oCU5WijvcyVbEODneHuLyA  0-7N3z_cb3Z1xMGDrbez6w   
160559  4mZYaPXdEXqKhZjWeVdT8w  xhAzk9qYpHR-nu_0sZIcWg   
165218  mmOSErjIXO87TskD8Gxspg  BcyHAag7lr2bVZGNW4u1dA   
174086  T1rmsie2h5VZ3prpXl-cMQ  zymvTQ12tRy3bzmP4tPgOg   
219811  6HcD_ikXQURCMbf5nZAZVg  0Nqa0zPHoB_GVucVMd9f6w   
226484  3CDy5wPg8mwqECGNU4geGQ  nzJoqZ4kSYM6ppg1EoLFVg   
228449  Mu3MfMUplft1DAoYsgj-9w  BrHFEGVof2yzU2JWD5aolw   
236599  nnWFrvhO2jeq0KbG_CKCYA  jmG_QxXXfz2cnw9dQCXJLQ   
236748  ajDPsSD77sxcVttPfftBRQ  DTxB12bQaZ1m-nLeXWX2rw   
245755  wYo3aBVj-bRPT7E4RSj5Kg  W5SNps2JaT_RozLAl_TN1Q   

                                                     text                date  \
10501                               love the crispy buns! 2017-10-18 23:29:03   
13633   Pricey, often crowded, and staff are not alway... 2017-10-20 22:08:25   
16776   Very friendly staff.  I really like this offic... 2017-09-27 18:29:38   
26341                      The i7information ase is in my 2017-09-12 20:19:36   
34982                     Delicious dinner at CHUlicious! 2017-08-02 00:22:58   
38060   Very good pizza. One of the best in Wilmington... 2017-11-03 22:13:04   
39703                              Great food, fun place! 2017-10-10 20:13:08   
64523        Love it. Very special, unique and delizioso! 2017-04-01 23:52:40   
69495   Totally in love with their croissants, pastrie... 2017-06-23 00:08:09   
83009   Hurry up and wait. The lines are too long! Hir... 2017-10-17 13:19:33   
137493                                 Excellent service! 2019-01-24 16:14:10   
149596                        Place is gone closed kaput. 2017-11-18 22:42:27   
149902                                    Slow........... 2017-09-24 04:28:50   
152067  Yummy ice cream and you get a ton of ice cream... 2017-10-01 22:23:59   
153167  Really wonderful food on my third visit. Still... 2017-10-21 02:06:33   
160559  Not as good as the NYC spots and the prices ar... 2017-07-03 22:55:35   
165218                                         Great food 2017-10-30 18:44:19   
174086  good ! it's refreshing to see a new great rest... 2019-08-07 23:45:25   
219811                          This place is really nice 2017-09-05 03:36:35   
226484  Unprofessional, unreliable, and untrustworthy ... 2019-12-14 03:14:03   
228449                                       Food is good 2021-05-15 15:48:42   
236599                    Great place, great baked goods! 2019-06-08 17:06:50   
236748                    Good service and delicious food 2017-09-12 23:01:50   
245755  Worker there are the best at serving me quick ... 2017-12-17 01:10:52   

        compliment_count  year  
10501                  0  2017  
13633                  0  2017  
16776                  0  2017  
26341                  0  2017  
34982                  0  2017  
38060                  0  2017  
39703                  0  2017  
64523                  0  2017  
69495                  0  2017  
83009                  0  2017  
137493                 0  2019  
149596                 0  2017  
149902                 0  2017  
152067                 0  2017  
153167                 0  2017  
160559            

In [ ]:
# elimina duplicados
df.drop_duplicates(inplace=True)
df.reset_index(drop=True, inplace=True) # reset index

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 286918 entries, 0 to 286917
Data columns (total 6 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   user_id           286918 non-null  object        
 1   business_id       286918 non-null  object        
 2   text              286918 non-null  object        
 3   date              286918 non-null  datetime64[ns]
 4   compliment_count  286918 non-null  int64         
 5   year              286918 non-null  int64         
dtypes: datetime64[ns](1), int64(2), object(3)
memory usage: 13.1+ MB


In [ ]:
# cantidad de tips por usuario
df['user_id'].value_counts().head()

fCvMnJU1Z-XhAjKg99wK3Q    3054
xWmYN57XXZbg0LOK8WbbFQ     813
sraTrYU-7q_bQ0TxQrLJvQ     627
6jjHo9Lilv3kTy87pm2ycw     470
sLgekUp1uedWpnDsztvkVw     447
Name: user_id, dtype: int64

In [ ]:
# cantidad de tips para cada business
df['business_id'].value_counts().head()

FEXhWNCMkv22qG04E83Qjg    933
ytynqOUb3hjKeJfRj5Tshw    332
ac1AeYqs8Z4_e2X5M3if2A    316
GXFMD0Z4jEVZBCsbPf4CTQ    303
_ab50qdWOk0DdB6XOrBitw    295
Name: business_id, dtype: int64

------------------

# checkin

---------------

In [ ]:
df = pd.read_parquet('/content/drive/MyDrive/DataYelpGoogle/checkin.parquet', engine='fastparquet')

In [ ]:
# info data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 131930 entries, 0 to 131929
Data columns (total 2 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   business_id  131930 non-null  object
 1   date         131930 non-null  object
dtypes: object(2)
memory usage: 2.0+ MB


In [ ]:
# show data
df.sample(5)

business_id  \
93491  hQV0OHY4d5l0tJKuLQB7tg   
3758   0l3rta_iJ8Jg59FD3v0huA   
19366  8HdSXgVPcZUBvaTKqlUL2A   
93783  hZvz3zAVoL2QW7nhJ-JDJA   
8967   3Gp9mIQn6x4fzTE5roUNvw   

                                                    date  
93491  2013-11-19 22:13:43, 2015-12-01 16:43:27, 2019...  
3758   2021-06-26 19:50:23, 2021-07-11 03:45:49, 2021...  
19366  2020-01-02 16:26:02, 2020-01-10 15:25:18, 2020...  
93783  2017-11-15 05:00:09, 2018-02-21 00:32:56, 2019...  
8967   2012-10-02 01:16:32, 2012-11-05 22:42:57, 2012...

In [ ]:
df['business_id'].nunique()

131930

In [ ]:
df['date'][0]

'2020-03-13 21:10:56, 2020-06-02 22:18:06, 2020-07-24 22:42:27, 2020-10-24 21:36:13, 2020-12-09 21:23:33, 2021-01-20 17:34:57, 2021-04-30 21:02:03, 2021-05-25 21:16:54, 2021-08-06 21:08:08, 2021-10-02 15:15:42, 2021-11-11 16:23:50'